ci_tools/process_results.py

```diff
@@ -10,25 +10,38 @@
                         help='Github step summary')
 args = parser.parse_args()
 
+all_codes = ['GL01', 'GL02', 'GL03', 'GL05', 'GL06', 'GL07', 'GL08', 'GL09',
+             'GL10', 'SS01', 'SS02', 'SS03', 'SS04', 'SS05', 'SS06', 'ES01',
+             'PR01', 'PR02', 'PR03', 'PR04', 'PR05', 'PR06', 'PR07', 'PR08',
+             'PR09', 'PR10', 'RT01', 'RT02', 'RT03', 'RT04', 'RT05', 'YD01',
+             'SA01', 'SA02', 'SA03', 'SA04', 'EX01']
+
 warning_codes = ['EX01', 'SA01']
 
 errors = {}
 warnings = {}
+parsing_errors = []
 with open(args.report, 'r', encoding='utf-8') as f:
     for line in f:
-        file_name, code, msg = line.split(':')
-        if code in warning_codes:
-            if file_name not in warnings:
-                warnings[file_name] = [msg]
-            else:
-                warnings[file_name].append(msg)
-        else:
-            if file_name not in errors:
-                errors[file_name] = [msg]
+        try:
+            file_name, code, msg = line.split(':', maxsplit=2)
+            if code not in all_codes:
+                parsing_errors.append(line)
+                continue
+            if code in warning_codes:
+                if file_name not in warnings:
+                    warnings[file_name] = [msg]
+                else:
+                    warnings[file_name].append(msg)
             else:
-                errors[file_name].append(msg)
+                if file_name not in errors:
+                    errors[file_name] = [msg]
+                else:
+                    errors[file_name].append(msg)
+        except ValueError:
+            parsing_errors.append(line)
```

Comment:
When is this exception raised?

Reply:
It is raised if `split(':')` fails to find the correct number of colons

Reply:
Like in this case?
```
>>> a, b, c = (1, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: not enough values to unpack (expected 3, got 2)
```
A comment line would be useful.

Reply:
Noted!